components/builder-notify/src/client.rs

```diff
@@ -0,0 +1,56 @@
+use crate::error::{Error,
+                   Result};
+use builder_core::http_client::{ACCEPT_APPLICATION_JSON,
+                                USER_AGENT_BLDR};
+use reqwest::{header::HeaderMap,
+              Body,
+              Client,
+              Response};
+use std::{collections::HashMap,
+          iter::FromIterator};
+
+#[derive(Clone, Default)]
+pub struct WebhookClient {
+    inner: Client,
+}
+
+impl WebhookClient {
+    pub fn new() -> Result<Self> {
+        let header_values = vec![USER_AGENT_BLDR.clone(), ACCEPT_APPLICATION_JSON.clone()];
+        let headers = HeaderMap::from_iter(header_values.into_iter());
+
+        let client = reqwest::Client::builder().default_headers(headers)
+                                               .build()?;
+
+        Ok(WebhookClient { inner: client })
+    }
+
+    pub async fn push(&self, url: &str, payload: &str) -> Result<Response> {
```

Comment:
Should `url` be a property of `WebHookClient` instead of an arg here? It would be odd to have one webhook posting to multiple urls right?